### PR TITLE
Propagate errors on ecliptical coordinates to equatorial (Fixes #89)

### LINKED ISSUE #89: Propagate errors in ecliptic coordinates to equatorial coordinates
For pulsars with coordinates defined in the ecliptic (ELONG and ELAT) the uncertainties are not propagated to equatorial coordinates.

### PR DESCRIPTION
**Pull request by cbassa**
This PR adds functionality to propagate the errors on ecliptical coordinates to equatorial coordinates. It uses a different algorithm to that in `psrcat` itself (see the `defineEquatorial` function in `defineParams.c`). It is not clear what the `psrcat` implementation does precisely.

This algorithm takes the errors in ecliptical coordinates and rotates them to the equatorial coordinate frame using the position angle between the North Ecliptical Pole and the North Equatorial Pole (taken from Jean Meeus' Astronomical Algorithms, 2nd edition, p. 100). Errors are corrected for `cos(ELAT)` and `cos(DECJ)` factors.

Comparison of the resulting errors on equatorial coordinates against the online PSRCAT results shows reasonable agreement.

### CODE REVIEW

**Comment by mattpitkin** [comment]
Hi @cbassa, thanks very much for this addition. Could you fix the minor PEP8 [whitespace issues](https://github.com/mattpitkin/psrqpy/pull/90#issuecomment-962582258) and then push the changes.

I don't see any other issues, so if the tests pass I'm happy to merge this and make a new minor release.

File: psrqpy/search.py [hunk 1]
```diff
@@ -1303,6 +1303,36 @@ def derived_equatorial(self):
         self.update(RAJnew, name='RAJ')
         self.update(DECJnew, name='DECJ')
 
+        ELONG_ERR = self.catalogue['ELONG_ERR']
```
Comment (mattpitkin) [inline psrqpy/search.py:1306]: I think you'll need to make this line:

```python
if 'ELONG_ERR' in self.columns:
    ELONG_ERR = self.catalogue['ELONG_ERR']
else:
    ELONG_ERR = np.full(self.catalogue_len, np.nan)
```

File: psrqpy/search.py [hunk 2]
```diff
@@ -1303,6 +1303,36 @@ def derived_equatorial(self):
...
+        ELAT_ERR = self.catalogue['ELAT_ERR']
```
Comment (mattpitkin) [inline psrqpy/search.py:1307]: Similarly, I think you'll have to make this line:

```python
if 'ELAT_ERR' in self.columns:
    ELAT_ERR = self.catalogue['ELAT_ERR']
else:
    ELAT_ERR = np.full(self.catalogue_len, np.nan)
```

File: psrqpy/search.py (hunk 1 above)
Comment (cbassa) [reply psrqpy/search.py]: Thanks for the suggestion. I hadn't come up with a solution to deal with missing values, but this would do the trick.

**Comment by cbassa** [comment]
Unfortunately the CI fails and it's unclear to me how to fix it. Any help would be appreciated.

**Comment by mattpitkin** [comment]
I'm taking a look now and will see what I can do

**Comment by mattpitkin** [comment]
If you download and apply this patch [0001-search.py-fix-tests.patch.txt](https://github.com/mattpitkin/psrqpy/files/7520181/0001-search.py-fix-tests.patch.txt) if should fix things. To apply the patch, download it into your psrqpy repo and run:
```bash
git am 0001-search.py-fix-tests.patch.txt
```

**Review by mattpitkin — APPROVED** [review]

**Comment by mattpitkin** [comment]
@cbassa Thanks again for this patch. I'll merge it now and create a new release.

**Pull request merged**